server/middleware/validInfo.js

```diff
@@ -0,0 +1,24 @@
+module.exports = function(req, res, next) {
+   const { firstName, lastName, userRole, userEmail, userSlack, userPassword, userGithub, userClassId ,username,userPhone,cyfCity,userDateOfBirth} = req.body;
+
+  function validEmail(userEmail) {
+    return /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(userEmail);
+  }
+
+  if (req.path === "/register") {
+    console.log(!userEmail.length);
+    if (![userEmail, username, userPassword].every(Boolean)) {
+      return res.json("Missing Credentials");
+    } else if (!validEmail(userEmail)) {
+      return res.json("Invalid Email");
+    }
+  } else if (req.path === "/login") {
```

Comment:
I don't think this will scale very well, having a single middleware `validInfo` for potentially every new route. You might be better performing some simple validation either at the top of each new route, or creating different validation middleware, in their own separate functions for each route - though I don't like that option as much as you have to look at the route/controller _and_ a separate file to see what validation is required.
I would be surprised if there isn't an existing node validation library that does this kind of thing for you too, so you don't have to reinvent the wheel!

Reply:
Yes ,tomorrow I will try to implement validation using express validator js .If it is successful I will remove this authentication file,